skimage/measure/profile.py

```diff
@@ -66,7 +67,15 @@ def profile_line(image, src, dst, linewidth=1,
     else:
         pixels = ndi.map_coordinates(image, perp_lines,
                                      order=order, mode=mode, cval=cval)
-    intensities = pixels.mean(axis=1)
+
+    if intensity_type == 'mean':
+        intensities = pixels.mean(axis=1)
+    elif intensity_type == 'max' or 'maximum':
+        intensities = pixels.max(axis=1)
+    elif intensity_type == 'min' or 'minimum':
+        intensities = pixels.min(axis=1)
+    elif intensity_type == 'sum' or 'total':
+        intensities = pixels.sum(axis=1)
```

Comment:
I'd prefer to keep the number of possible values for `intensity_type` to a minimum. I'd omit `'max'` in favor of `'maximum'`, `'min'` in favor of `'minimum'` and `'sum'` in favor of `'total'`. Is there a reason you didn't add the median as a possible option?

Reply:
Oh and we should raise a ValueError in case `intensity_type` contains none of those values.

Reply:
@PinkShnack this is a great enhancement! But I think a much better implementation here is to forget about strings and allow passing any ufunc. I would call the argument (which yes, should be keyword-only) `aggregation=np.mean` or `summary_function=np.mean`, or a similar name, and then people can pass in any function that can take an array and an axis argument, and get the result:

```python
try:
    intensities = aggregation(pixels, axis=1)
except TypeError:  # function doesn't allow axis kwarg
    intensities = np.apply_along_axis(aggregation_function, pixels, axis=1)
```

Then the usage would be `profile_line(image, src, dst, aggregation=np.max)`

What do people think?

Reply:
You're right, that's way better! ^^ 

I still find `merge_channels` to be a better description of the parameter though. It makes it immediately obvious that it has no effect for images with only one channel (e.g. grayscale, in which case it only serves to remove an unnecessary axis of length 1 in `pixels`).

Reply:
Thanks for the quick responses all!

@lagru 
**1. Greyscale.** My reason for doing this enhancement was to change how the outputted pixels were dealt when summing/averaging etc., over a certain `linewidth`. For example, if you have a `profile_line` with a `linewidth` of 5, you will get different outputs when you look at the max, mean, min, median, or summed values of those 5 values along the `profile_line`. I hadn't even thought of multi channeled images, I deal solely with greyscale!

Therefore, should we have two different parameters? One for choosing the aggregation over the linewidth, and one for choosing the aggregation over multi-channel images?
Of course, the former will have no effect over a `profile_line` with `linewidth=1`. As far as I can see, these should be different parameters. Thoughts?

2. max, min, sum are simpler alright! No reason to leave out median let's include it! Any others?

**3. Keyword only argument.** I thought it should have a default because I don't want to interrupt people's old code!

@jni 

**ufunc.** That's great, so cool, I'll start using that in my code!

Reply:
Oh, I made two mistakes: I overlooked the `linewidth` argument and assumed that the mean function is used to reduce the dimension with different channels which it does not! That also makes my argument that the new keyword should relate to channels moot. As @jni suggested `aggregation` may be a good name, though now I particular like `reduce_func` which I find more explicit than the former.

> Therefore, should we have two different parameters? One for choosing the aggregation over the linewidth, and one for choosing the aggregation over multi-channel images?
Of course, the former will have no effect over a profile_line with linewidth=1. As far as I can see, these should be different parameters. Thoughts?

No I'd keep it at one parameter to control the reduction operation. As you correctly stated the old behavior is only relevant when `linewidth` is > 1. I see this mainly as a convenience argument to keep the output to one dimension. The user can always perform an additional reduction on multichannel images.

> 3. Keyword only argument. I thought it should have a default because I don't want to interrupt people's old code!

Did you perhaps misunderstand me? Being keyword-only does not preclude using a default value. With [keyword-only](https://www.python.org/dev/peps/pep-3102/) I mean parameter names that come after an `*,`, e.g.

```python
def foo(x, *, keyword_only=None):
    print(x, keyword_only)

foo(1, 2)
# "TypeError: foo() takes 1 positional argument but 2 were given"
```

Reply:
I like `reduce_func` as argument name ;-)

Reply:
FWIW, PyTorch uses `reduction`.